pkg/logging/http.go

```diff
@@ -24,17 +24,17 @@ type HTTPServerMiddleware struct {
 }
 
 func (m *HTTPServerMiddleware) preCall(name string, start time.Time, r *http.Request) {
-	logger := m.opts.filterLog(m.logger)
-	level.Debug(logger).Log("http.start_time", start.String(), "http.method", fmt.Sprintf("%s %s", r.Method, r.URL), "http.request_id", r.Header.Get("X-Request-ID"), "thanos.method_name", name, "msg", "started call")
+	m.opts.levelledLogger(m.opts.filterLog(m.logger)).Log(
```

Comment:
this logger could probably be configured once in the constructor?

Reply:
thinking about this some more I am also not sure that the filter is even needed considering we have setup a logger via `cmd/main.go` that is filtered based on `--log.level` and passed in to this middleware

Reply:
I think if we use the `leveledLogger` here to achieve the intended effect (i.e. log the requests on the configured level), `filterLog` should not be necessary as you said - we should have that governed by the global flag.

Reply:
For sure, totally agree - maybe @bwplotka could chime in and let us know if there were "bigger plans" for the logging package or do we really need to expose these functional options at all?